### ISSUE TITLE
HDR Option not showed

### ISSUE THREAD

**Issue by Buccolieri96** [zearp/Nucintosh#77]
Hello, I have an NUC8i3BEH with an BenQ EW3270U 4k HDR Monitor

I can't enable HDR in option and it's not showed. The monitor is connected with an HDMI to HDMI 2.0 cable, anyway, the 4k 60Hz signal is working well

**Comment by SeaDaddytheRed** [comment]
I had the same issue.  I don't think it's specific to our NUCs or Hackintoshs but a larger issue with Macs and HDMI.  I ended up getting a USB-C to Display Port cable and the colors and refresh rates are not longer a problem for me.

**Comment by zearp** [comment]
I don't have any monitors with HDR to test. As far as I understand there is not much we can do. Either macOS detects the screen and give the options to enable/disable HDR or it doesn't. For me to test these kind of things I would need to buy several HDR screens to test.

I would suggest trying SeaDaddytheRed's suggestion. Either with a usb-c hub that has a DisplayPort or a usb-c to DisplayPort cable. There also some patches and other things to try and force the screen to enable full HDR/RGB modes. Hackintool also has some of those. It will require some experimentation to get it going.

For me there's not much I Can do other than buy every piece of hardware that isn't working fully. But I lack the time and funds to do so.

**Comment by zearp** [comment]
I don't have any monitors with HDR to test. As far as I understand there is not much we can do. Either macOS detects the screen and give the options to enable/disable HDR or it doesn't. For me to test these kind of things I would need to buy several HDR screens to test.

I would suggest trying SeaDaddytheRed's suggestion. Either with a usb-c hub that has a DisplayPort or a usb-c to DisplayPort cable. There also some patches and other things to try and force the screen to enable full HDR/RGB modes. Hackintool also has some of those. It will require some experimentation to get it going.

For me there's not much I can do other than buy every piece of hardware that isn't working fully. But I lack the time and funds to do so.

**Comment by felixg3** [comment]
I can confirm that HDR does not work with either DisplayPort via Type-C nor HDMI.

**Comment by zearp** [comment]
I can confirm the HDR option does not show in Monterey, Big Sur and also Catalina. I've tested it with a screen that works fine with HDR in Windows.

From reading up on the issue to get HDR to work you may need to run Big Sur oir Catalina and also use a usb-c to displayPort converter. Not all converters will work.

There are some thread about this on several Mac user forums who run into the same issue; no HDR on their 3rd party display. some got it to work by using a $60 (!!) Belkin adapter. I think that one uses Thunderbolt and not the usb protocol. Which would explain the price and why it works. Some other reported success with cheaper adapters but I haven't found any of those for sale myself.

It might be an idea to oder a bunch of them from Amazon, test them all and make a list of which ones will actually provide HDR options. I think this is one of these weird Apple only things where some small detail makes all the difference. I have a Thunderbolt 2 hub somewhere with DisplayPorts so I can try it but it needs a $50 cable to convert the hub to Thunderbolt 3. If Amazon sells it I can try it and return it as I'm not planning on using it other than testing.

From what I read Monterey needs a lot of HDR fixes. For many Mac users HDR is completely broken in Netflix and Youtube and such. I would advise to do any testing in either Big Sur or Catalina just to rule out the Monterey HDR issues.

Another thing I noticed is that Catalina is the only macOS that detected all my refresh rates properly. The only good news is that 10bit colours seem to work out of the box on all 3 versions. Monterey and Big Sur only went up to 144hz while Catalina could also use the 165hz refresh rate. We can force higher refresh rates manually but it's telling how badly Apple is supporting 3rd party displays on newer versions.

Long story short, I have no idea how to enable HDR without buying a bunch of stuff to try things Mac users also tried. I also wonder if HDR works if I connect up an Apple display but I don't own any of those.

I will be trying and testing some more things in the days to come but I'm not confident it can be made work as so many Mac users struggle with the same issue.

**Comment by zearp** [comment]
I can confirm the HDR option does not show in Monterey, Big Sur and also Catalina. I've tested it with a screen that works fine with HDR in Windows.

From reading up on the issue to get HDR to work you may need to run Big Sur or Catalina and also use a usb-c to DisplayPort converter. Not all converters will work. Some forum users said a HDMI converter did work for them too. Apple does support DisplayPoort better than HDMI and none the HDMI ports I have on my usb-c hubs worked. I tried 3 of them all using different chipsets.

There are some threads about this issue on several Mac forums from people who run into the same issue; no HDR on their 3rd party display. Some got it to work by using a $60 (!!) Belkin adapter. I think that one uses Thunderbolt and not the usb protocol. Which would explain the price and why it works. Some others reported success with cheaper adapters but I haven't found any of those for sale myself.

It might be an idea to order a bunch of these converters from Amazon, test them all and make a list of which ones will actually provide HDR options and which ones don't. I think this is one of these weird Apple only things where some small detail makes all the difference. I have a Thunderbolt 2 hub somewhere with DisplayPorts so I can test if Thunderbolt would help but it needs a $50 cable to convert the hub to Thunderbolt 3. If Amazon sells it I can try it and return it, but I'm not planning on keeping it.

From what I read Monterey needs a lot of HDR fixes. For many Mac users HDR is completely broken in Netflix and Youtube and such. I would advise to do any testing in either Big Sur or Catalina just to rule out the Monterey HDR issues.

Another thing I noticed is that Catalina was the only macOS that detected all my refresh rates properly.

Some good news is that 10bit colours seem to work out of the box on all 3 versions. Monterey and Big Sur only went up to 144hz while Catalina could also use the 165hz refresh rate. We can force higher refresh rates manually but it's telling how badly Apple is supporting 3rd party displays on newer macOS versions.

Long story short, I have no idea how to enable HDR without buying a bunch of stuff to try things Mac users also tried. I also wonder if HDR works if I connect up an Apple display but I don't own any of those.

I will be trying and testing some more things in the days to come but I'm not confident it can be made work as so many Mac users struggle with the same issue.

**Comment by zearp** [comment]
Amazon Basics bi-directional usb-c to DisplayPort cable does not show the HDR option.

**Comment by zearp** [comment]
DolbyVision compatible cables start around $10, will try some of those too!

**Comment by zearp** [comment]
Today's usb-c to DisplayPort 1.4 converter also did not provide any HDR option. Tomorrow a cable sold as Dolby Vision compatible arrives. If that cable does not work I will move to the premium ones starting with CalDigit's adapter.

**Comment by zearp** [comment]
Also no luck with a Dolby Vision cable. Maybe it was the cable but I'm starting to think Monterey HDR is not possible. I have no tried any of the cables in earlier macOS. I might try when I got more spare time.

**Comment by zearp** [comment]
Also no luck with a Dolby Vision cable. Maybe it was the cable but I'm starting to think Monterey HDR is not possible. I have not tried any of the cables in earlier macOS. I might try when I got more spare time.

**Comment by zearp** [comment]
I got the HDR option to show with the Amazon Basics bi-directional cable by not hot swapping them but cold booting with it. IT will only make the box appear. Enabling it causes the colours to wash out and the screen OSD doesn't report any HDR signal. Playing HDR content on Youtube does not work either.

I'll retest some other usb-c cables this way. DisplayPort stuff often does not like hotplug events much. I hope HDR can be made to work without having to spend $50 on some "approved" converter.

![Screenshot 2022-07-14 at 02 06 24](https://user-images.githubusercontent.com/63272687/178857679-251f40e2-ca78-4251-863a-edcd42970aaa.png)

**Comment by SeaDaddytheRed** [comment]
I have a Cable Matters USB-C to DisplayPort going to an LG monitor.  I also have the HDR option in display (both under Monterey and Ventura) but when I enable it I get the washed out color and the monitor does not report HDR in use.

**Comment by zearp** [comment]
I had the same with all usb-c cables I tested, they all result in washed out colours and the screen never sees any HDR signal despite macOS claiming it's enabled. Additionally 10bit colours seems to be impossible. Which to me is worse than not having HDR. Maybe the tool I used isn't telling me the truth but I think 10bit colours also is not working. I will need to do more testing.
```
  mode 12: res:1920x1080 hz:50 color_depth:4
  mode 13: res:1920x1080 hz:50 color_depth:8
  mode 14: res:1920x1080 hz:120 color_depth:4
  mode 15: res:1920x1080 hz:120 color_depth:8 <-- current mode
  mode 16: res:1920x1080 hz:100 color_depth:4
  mode 17: res:1920x1080 hz:100 color_depth:8
  mode 18: res:1920x1080 hz:120 color_depth:4
  mode 19: res:1920x1080 hz:120 color_depth:8
  mode 20: res:1920x1080 hz:144 color_depth:4
  mode 21: res:1920x1080 hz:144 color_depth:8
```
These are the only 1080p modes available to me in Monterey. In earlier macOS version I also have 165hz but no macOS version I tried showed any 10bit colour modes. Any HDR screen is capable of doing 10bit colours. It kind of sucks I have to boot back into Windows to enjoy HDR and 10bit colours. So I really hope this can be made to work properly.

Tool used to get available resolutions/modes: https://github.com/jakehilborn/displayplacer

**Comment by zearp** [comment]
System profiler shows Framebuffer depth as 30bit colours, which is the same as 10bit. I don't know what to believe. I will have a dig with IORegistryExplorer too and see what kind of info that gives me. I'll also get some 10bit test images with gradients and such. It shouldn't be too difficult to tell if its 8bit or 10bit. When it's really 10bit there will be much less colour banding.
```
  Displays:
  Resolution:	1920x1080 (1080p FHD - Full High Definition)
  UI Looks like:	1920 x 1080 @ 120.00Hz
  Framebuffer Depth:	30-Bit Colour (ARGB2101010)
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Connection Type:	DVI or HDMI
  Adapter Firmware Version:	7,55
```

**Comment by zearp** [comment]
I did some testing on Catalina and I can confirm that HDR works in Catalina with downloaded videos. Confirmed this by watching a video and checking the monitor OSD information. I tried in Safari with Youtube but there was no HDR option visible. I will do some more testing but I don't think streaming media in HDR will work due to lack of proper DRM. Consuming HDR media should work in Catalina as long as you don't stream it.

Catalina still gets security updates and it might be the only way to enjoy HDR content in macOS on our little NUCs. I don't think there's some magic trick or setting we can use that fixes the washed our colours and HDR not really being enabled in Big Sur and newer. Even people who own modern Mac hardware have issues with HDR on 3rd party screens.

At least HDR will work in Catalina if you're using a decent cable. I've tested the Amazon Basics bi-directional usb-c to. display port cable. It should be cheap and easy to get anywhere. The downside is you lose the usb-c port. Other cables might work too in Catalina. I can't test it anymore as I've returned all the cables except the Amazon Basics ones. Amazon refunded me and let me keep it when I made a return request for it.
